plugin/storage/es/spanstore/service_operation.go

```diff
@@ -53,17 +57,28 @@ func NewServiceOperationStorage(
 }
 
 // Write saves a service to operation pair.
-func (s *ServiceOperationStorage) Write(indexName string, jsonSpan *dbmodel.Span) {
+func (s *ServiceOperationStorage) Write(indexName string, jsonSpan *dbmodel.Span, kind model.SpanKind) {
 	// Insert serviceName:operationName document
 	service := dbmodel.Service{
 		ServiceName:   jsonSpan.Process.ServiceName,
 		OperationName: jsonSpan.OperationName,
 	}
-
-	cacheKey := hashCode(service)
-	if !keyInCache(cacheKey, s.serviceCache) {
-		s.client().Index().Index(indexName).Type(serviceType).Id(cacheKey).BodyJson(service).Add()
-		writeCache(cacheKey, s.serviceCache)
+	if kind != model.SpanKindUnspecified {
```

Comment:
why do we need to bifurcate? If for some reason span kind is unavailable, we can just treat it as blank string and not change the code to deal with it separately.

Reply:
Please the comment https://github.com/jaegertracing/jaeger/pull/6399#issuecomment-2562068739. If the reason seems valid then I don't think we should introduce empty kinds because that will become messy along with data of `Without Kind` (old data) and then in reader it might become more complex while fetching.